nitweb: add graphs stubs

Morriar opened the pull request:
Add some very simple graphs tabs to nitweb:

* Packages: http://nitweb.moz-code.org/package/popcorn
* Groups: http://nitweb.moz-code.org/group/popcorn%3E
* Modules: http://nitweb.moz-code.org/module/core::array
* Classes: http://nitweb.moz-code.org/class/core::Array

ventilooo commented:
+1 However, when there is a large number of inheritance or dependance the graph can become a mess. Is that possible to have less curvy graph ?

privat commented on src/web/model_api.nit:
```diff
@@ -166,11 +182,76 @@ class APIEntityDefs
+				op.append("\"{child.name.escape_to_dot}\"->\"{center.name.escape_to_dot}\";\n")
+			end
+		else
+				op.append("\"...\"->\"{center.name.escape_to_dot}\";\n")
```
you should use the fullname for identifying the nodes

privat commented:
+1

jcbrinfo commented:
:+1: +1

But, please change the font if you can: Times is so badly kerned that `popcorn` is displayed as `popcom` (for example). :smile:

Tagachi commented:
The new interface is pretty cool :)
However, with angular the web page load before his content. Maybe, you can add a gif to highlight : content is comming

Morriar commented:
Major reroll of this PR:
* switched to a minimalist dot library
* added some heuristics to improve graph lisibility

Some examples:
* package dependencies: http://nitweb.moz-code.org/package/mongodb
* group importation: http://nitweb.moz-code.org/group/test_suite%3E
* module importation: http://nitweb.moz-code.org/module/json::json
* class inheritance: http://nitweb.moz-code.org/class/core::Text

One the API side, the parents/children depth can be given as argument:
* http://nitweb.moz-code.org/api/graph/inheritance/core::Text (no args, full depth)
* http://nitweb.moz-code.org/api/graph/inheritance/core::Text?cdepth=1 (children depth)
* http://nitweb.moz-code.org/api/graph/inheritance/core::Text?pdepth=2 (parent depth)
* http://nitweb.moz-code.org/api/graph/inheritance/core::Text?pdepth=1&cdepth=2

R4PaSs commented:
These look nice !

Having links on the entities is a nice plus, although they seem broken, at least on my machine.
The target of the link looks OK though, I'm not sure why I'm sent back to the index when clicking on an entity.

+1 !

Morriar commented:
@R4PaSs Can you try to purge your cache?

R4PaSs commented:
Yep, no changes on my part with a clear cache it seems

jcbrinfo commented on src/web/api_graph.nit:
```diff
@@ -0,0 +1,273 @@
+		graph.nodes_attrs["width"] = 0
+		graph.nodes_attrs["height"] = 0
+		graph.nodes_attrs["fontsize"] = 10
+		graph.nodes_attrs["fontname"] = "helvetica"
```
Could you put just `sans-serif` (like the rest of the UI) or `monospace`, or clear it? I do not know why, but when my web browser sees an unknown font in the `font-family` SVG attribute, it fallbacks to an ugly default.

Here an example with `font-family="Helvetica,sans-Serif"` (what nitweb generates):
![ng-doc_-_2016-06-21_11 46 55](https://cloud.githubusercontent.com/assets/6044484/16236716/573985a0-37a7-11e6-8a08-7fd71cc2823b.png)


If I change it to `font-family="sans-serif"`, `font-family="sans-serif"` or remove the attribute:
![ng-doc_-_2016-06-21_11 52 46](https://cloud.githubusercontent.com/assets/6044484/16236733/6be35d1e-37a7-11e6-8682-f96f68cb00f8.png)

If  I change it to `font-family="monospace"`:
![ng-doc_-_2016-06-21_11 56 35](https://cloud.githubusercontent.com/assets/6044484/16236739/70a616ca-37a7-11e6-9a38-c57170f6532a.png)

jcbrinfo commented:
The new diagrams look more like UML. The only thing missing are the arrow ends. :wink: 

I reiterate my :+1: +1

privat commented:
+1
